### PR TITLE
fixes for feedback from qa

### PR DESCRIPTION
**Pull request by deeonwuli**
### :pushpin: References

-   **Issue:** Closes 
- https://app.clickup.com/t/1mufxfj
- https://app.clickup.com/t/1crue0m

### CODE REVIEW

File: src/webapp/pages/bulk-apply/steps/AccessStep.tsx [hunk 1]
```diff
@@ -59,13 +61,37 @@ export const AccessStep: React.FC<MetadataSharingWizardStepProps> = ({ builder,
...
+            </ConfirmationDialog>
+
             <h4>Advanced options</h4>
             <FormControlLabel
                 control={
                     <Switch
                         name="replace-existing-sharings"
                         checked={builder.replaceExistingSharings}
-                        onChange={setReplaceStragegy}
+                        onChange={e =>
```
Comment (anagperal) [inline src/webapp/pages/bulk-apply/steps/AccessStep.tsx]: This change works, but onChange function returns (event: React.ChangeEvent<HTMLInputElement>, isReplacingExistingSharings: boolean), so it's better to maintain this logic:

```typescript
      onChange={(event: React.ChangeEvent<HTMLInputElement>, isReplacingExistingSharings: boolean) =>
                            isReplacingExistingSharings
                                ? setDialogOpen(true)
                                : setReplaceStragegy(event, isReplacingExistingSharings)
                        }
```

File: src/webapp/pages/bulk-apply/steps/ListDependenciesStep.tsx [hunk 2]
```diff
@@ -137,13 +162,27 @@ export const ListDependenciesStep: React.FC<MetadataSharingWizardStepProps> = ({
...
-        />
+        <>
+            <Dropdown<MetadataModel>
+                items={filterOptions}
+                onValueChange={applyFilterChanges}
+                value={listOptions.model}
+                label={i18n.t("Metadata type")}
+                hideEmpty={true}
+            />
+            <Button
```
Comment (anagperal) [inline src/webapp/pages/bulk-apply/steps/ListDependenciesStep.tsx]: Add same margin-left that the Dropdown has to make it look better

File: src/webapp/pages/bulk-apply/steps/SummaryApplyStep.tsx [hunk 3]
```diff
@@ -26,7 +28,24 @@ export const SummaryApplyStep: React.FC<MetadataSharingWizardStepProps> = ({ bui
         <React.Fragment>
             {importResult && <ImportSummary results={[importResult]} onClose={() => setImportResult(undefined)} />}
 
-            <Button variant="contained" color="primary" onClick={applySharingSync}>
+            <ConfirmationDialog
+                isOpen={openDialog}
+                title={i18n.t("Warning")}
+                onCancel={() => setDialogOpen(false)}
+                onSave={applySharingSync}
```
Comment (anagperal) [inline src/webapp/pages/bulk-apply/steps/SummaryApplyStep.tsx:44]: onSave execute close this dialog and applySharingSync (in a useCallback hook)

**Review by anagperal — APPROVED** [review]
-

**Review by anagperal — CHANGES_REQUESTED** [review]
thanks @deeonwuli ! Apart from the comments in the code, accoding to taks in https://app.clickup.com/t/1mufxfj
- some task in the checklist are not done. I don't know if it's because it has to be that way and there is no need for them to be made
- Add pagination to bottom of table (misc, this is normal behavior everywhere) and Selection reloads order and moves (ENTO- vs ENTO -) [See GIF ss-1] are not implemented

And according with task https://app.clickup.com/t/1crue0m Warnings page with Users/Groups assigned to parent not assigned to children (Both meta and data shared) is not implemented (I don't know if it's because it has to be that way and there is no need)

**Comment by MiquelAdell** [comment]
@anagperal sorry because I should have access to that information but… What checklist are you referencing in your previous comment? 
As I want to release a version of this app Monday at the latest if the functionality for "Warnings page with Users/Groups assigned to parent not assigned to children" is not present I will create a new task for that and merge the current changes.
Most provably the same will be true for this task in the checklist.

TL;DR: finish started tasks that are failing, I'll create new tasks for pending code

**Comment by anagperal** [comment]
> @anagperal sorry because I should have access to that information but… What checklist are you referencing in your previous comment? As I want to release a version of this app Monday at the latest if the functionality for "Warnings page with Users/Groups assigned to parent not assigned to children" is not present I will create a new task for that and merge the current changes. Most provably the same will be true for this task in the checklist.
> 
> TL;DR: finish started tasks that are failing, I'll create new tasks for pending code

@MiquelAdell this is the checklist:
image

It's in actions item tab in https://app.clickup.com/t/1mufxfj

**Comment by MiquelAdell** [comment]
Thanks @anagperal I've moved both action items to their own tasks. As well as "Add pagination to bottom of table…" to a new task.

@deeonwuli I'm not sure if this task is blocked by something or we just need to fix some bug regarding it:
https://app.clickup.com/t/1crue0m

**Review by anagperal — CHANGES_REQUESTED** [review]
This comment is not solved yet: https://github.com/EyeSeeTea/sharing-settings-app-dev/pull/11#discussion_r1579111679

**Review by anagperal — APPROVED** [review]
@deeonwuli code looks good, thanks!
